src/metpy/calc/tools.py

```diff
@@ -1946,3 +1947,105 @@ def _remove_nans(*variables):
     for v in variables:
         ret.append(v[~mask])
     return ret
+
+
+def _neighbor_inds(y, x):
+    """Generate index (row, col) pairs for each neighbor of (x, y)."""
+    incs = (-1, 0, 1)
+    for dx, dy in itertools.product(incs, incs):
+        yield y + dy, x + dx
+
+
+def _find_uf(uf, item):
+    """Find the root item for ``item`` in the union find structure ``uf``."""
+    # uf is a dictionary mapping item->parent. Loop until we find parent=parent.
+    while (next_item := uf[item]) != item:
+        uf[item] = uf[next_item]
+        item = next_item
+    return item
+
+
+@exporter.export
+def persistence(arr, *, maxima=True):
```

Comment:
Still doing my own quick once-over on implementation, but a thought: if this is public, "persistence" is a term already prevalent in forecasting. This could be a point of minor confusion for our base of users.

Reply:
Good point, though not sure what that would mean as a calculation. `peak_persistence`? Or do we just avoid making it public right now?

Reply:
It wouldn't mean much to anyone at the point of use. I think it would only be minorly confusing at the point of searching docs or listing available calcs. I think it being public is helpful. `peak/object/feature/extrema_persistence` is fine by me here.

Reply:
Ok, changed to `peak_persistence`.